common/FileUtils.java

```diff
@@ -238,19 +241,36 @@ public Message readProtoBinaryUnchecked(String path, Message.Builder builder) {
    * Copies a directory to another directory. Creates target directory if needed and doesn't copy
    * symlinks.
    */
-  public void copyDirectoryToDirectory(String source, String destination, String ignored)
+  public void copyDirectoryToDirectory(String source, String destination, String... ignored)
       throws IOException {
     final Path sourcePath = fileSystem.getPath(source);
     final Path targetPath = fileSystem.getPath(destination);
+
+    List<String> allFilesAndFoldersForIgnore = new ArrayList<>();
+    if (ignored.length != 0) {
+      for (String itemForIgnore : ignored) {
+        String itemForIgnorePath = joinPaths(source, itemForIgnore);
+        if (folderExists(itemForIgnorePath) || fileExists(itemForIgnorePath)) {
+          allFilesAndFoldersForIgnore.addAll(getListContentsWithAbsolutePath(itemForIgnorePath));
```

Comment:
Why do we need `getListContentsWithAbsolutePath()`? Can you give an example?

Reply:
I deleted this method and simplify the code.